### DISCUSSION TITLE
primary_key=True in Field not work when sa_column argument exists

### DISCUSSION THREAD

**Discussion by honglei** [fastapi/sqlmodel#573] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
##Not work 
class ResourceType(SQLModel, table=True):
    resourceTypeID: int = Field(sa_column=Column(SmallInteger),primary_key=True
                                 index=True, ) 
# Work
class ResourceTypeOK(SQLModel, table=True):
    resourceTypeID: int = Field(sa_column=Column(SmallInteger,primary_key=True)
                                 index=True, )
```

### Description

Error: sqlalchemy.exc.ArgumentError: Mapper Mapper[ResourceType(resourcetype)] could not assemble any primary key columns for mapped table 'resourcetype'

### Operating System

Windows

### SQLModel Version

0.0.8

### Python Version

Python 3.11.2

**Comment by mavwolverine** [comment]
On similar lines

```
class Item(SQLModel, table=True):
    id: Optional[int] = Field(default=None, sa_column=mapped_column('id', Integer, primary_key=True))
    # {Column('id', Integer(), table=<item>)}
    # primary_key is not set when the column reaches sqlalchemy

    # id: Optional[int] | None = Field(default=None, primary_key=True)
    # {Column('id', Integer(), table=<item>, primary_key=True, nullable=False)}
```

**Comment by YuriiMotov** [comment, marked as answer]
Yes, if you specify `sa_column` you should then pass `primary_key=True`, `index=True` and other column-related parameters as parameters to `Column`:
```py
class ResourceTypeOK(SQLModel, table=True):
    resourceTypeID: int = Field(
        sa_column=Column(SmallInteger, primary_key=True, index=True),
    )
```
